Azure sdk update

mwilsoninsight opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#sec-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

All modules build without error in python 3.7

kalbasit commented:
The commit messages do not follow [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

mwilsoninsight commented:
> The commit messages do not follow [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Yep- sorry about that. Originally I wanted to save on the number of commits, but that doesn't quite work for following the contributing guide.

Thanks!

kalbasit commented on pkgs/development/python-modules/azure-servicemanagement-legacy/default.nix:
```diff
@@ -1,31 +1,43 @@
+  # has no tests
+  doCheck = false;
+
+  meta = lib; {
```
you're missing `with` in `with lib` here, and it's breaking the eval.

mwilsoninsight replied:
0.o
I've even fixed this before... many thanks kalbasit!

I just git rebase --interactive'd my way into adding the necessary `with` 👍

kalbasit commented:
Alright, let's do first Python2 :crossed_fingers: 

@GrahamcOfBorg build python2Packages.azure-applicationinsights python2Packages.azure-batch python2Packages.azure-common python2Packages.azure-cosmosdb-nspkg python2Packages.azure-cosmosdb-table python2Packages.azure-datalake-store python2Packages.azure-eventgrid python2Packages.azure-graphrbac python2Packages.azure-keyvault python2Packages.azure-loganalytics python2Packages.azure-mgmt-advisor python2Packages.azure-mgmt-applicationinsights python2Packages.azure-mgmt-authorization python2Packages.azure-mgmt-batch python2Packages.azure-mgmt-batchai python2Packages.azure-mgmt-billing python2Packages.azure-mgmt-cdn python2Packages.azure-mgmt-cognitiveservices python2Packages.azure-mgmt-commerce python2Packages.azure-mgmt-compute python2Packages.azure-mgmt-consumption python2Packages.azure-mgmt-containerinstance python2Packages.azure-mgmt-containerservice python2Packages.azure-mgmt-cosmosdb python2Packages.azure-mgmt-datafactory python2Packages.azure-mgmt-datalake-analytics python2Packages.azure-mgmt-datalake-nspkg python2Packages.azure-mgmt-datalake-store python2Packages.azure-mgmt-datamigration python2Packages.azure-mgmt-devspaces python2Packages.azure-mgmt-devtestlabs python2Packages.azure-mgmt-dns python2Packages.azure-mgmt-eventgrid python2Packages.azure-mgmt-eventhub python2Packages.azure-mgmt-hanaonazure python2Packages.azure-mgmt-iotcentral python2Packages.azure-mgmt-iothub python2Packages.azure-mgmt-iothubprovisioningservices python2Packages.azure-mgmt-keyvault python2Packages.azure-mgmt-loganalytics python2Packages.azure-mgmt-logic python2Packages.azure-mgmt-machinelearningcompute python2Packages.azure-mgmt-managementgroups python2Packages.azure-mgmt-managementpartner python2Packages.azure-mgmt-maps python2Packages.azure-mgmt-marketplaceordering python2Packages.azure-mgmt-media python2Packages.azure-mgmt-monitor python2Packages.azure-mgmt-msi python2Packages.azure-mgmt-network python2Packages.azure-mgmt-notificationhubs python2Packages.azure-mgmt-policyinsights python2Packages.azure-mgmt-powerbiembedded python2Packages.azure-mgmt-rdbms python2Packages.azure-mgmt-recoveryservices python2Packages.azure-mgmt-recoveryservicesbackup python2Packages.azure-mgmt-redis python2Packages.azure-mgmt-relay python2Packages.azure-mgmt-reservations python2Packages.azure-mgmt-resource python2Packages.azure-mgmt-scheduler python2Packages.azure-mgmt-search python2Packages.azure-mgmt-servicebus python2Packages.azure-mgmt-servicefabric python2Packages.azure-mgmt-signalr python2Packages.azure-mgmt-sql python2Packages.azure-mgmt-storage python2Packages.azure-mgmt-subscription python2Packages.azure-mgmt-trafficmanager python2Packages.azure-mgmt-web python2Packages.azure-servicebus python2Packages.azure-servicefabric python2Packages.azure-servicemanagament-legacy python2Packages.azure-storage-blob python2Packages.uamqp

mwilsoninsight commented:
I had a typo in keyvault =/

Just force pushed the fix

mwilsoninsight commented:
I see the same typo in service{bus,fabric} and uamqp- pushing those fixes shortly

mwilsoninsight commented:
> ~Alright, let's do first Python2 crossed_fingers~
> 
> ~@GrahamcOfBorg build python2Packages.azure-applicationinsights python2Packages.azure-batch python2Packages.azure-common python2Packages.azure-cosmosdb-nspkg python2Packages.azure-cosmosdb-table python2Packages.azure-datalake-store python2Packages.azure-eventgrid python2Packages.azure-graphrbac python2Packages.azure-keyvault python2Packages.azure-loganalytics python2Packages.azure-mgmt-advisor python2Packages.azure-mgmt-applicationinsights python2Packages.azure-mgmt-authorization python2Packages.azure-mgmt-batch python2Packages.azure-mgmt-batchai python2Packages.azure-mgmt-billing python2Packages.azure-mgmt-cdn python2Packages.azure-mgmt-cognitiveservices python2Packages.azure-mgmt-commerce python2Packages.azure-mgmt-compute python2Packages.azure-mgmt-consumption python2Packages.azure-mgmt-containerinstance python2Packages.azure-mgmt-containerservice python2Packages.azure-mgmt-cosmosdb python2Packages.azure-mgmt-datafactory python2Packages.azure-mgmt-datalake-analytics python2Packages.azure-mgmt-datalake-nspkg python2Packages.azure-mgmt-datalake-store python2Packages.azure-mgmt-datamigration python2Packages.azure-mgmt-devspaces python2Packages.azure-mgmt-devtestlabs python2Packages.azure-mgmt-dns python2Packages.azure-mgmt-eventgrid python2Packages.azure-mgmt-eventhub python2Packages.azure-mgmt-hanaonazure python2Packages.azure-mgmt-iotcentral python2Packages.azure-mgmt-iothub python2Packages.azure-mgmt-iothubprovisioningservices python2Packages.azure-mgmt-keyvault python2Packages.azure-mgmt-loganalytics python2Packages.azure-mgmt-logic python2Packages.azure-mgmt-machinelearningcompute python2Packages.azure-mgmt-managementgroups python2Packages.azure-mgmt-managementpartner python2Packages.azure-mgmt-maps python2Packages.azure-mgmt-marketplaceordering python2Packages.azure-mgmt-media python2Packages.azure-mgmt-monitor python2Packages.azure-mgmt-msi python2Packages.azure-mgmt-network python2Packages.azure-mgmt-notificationhubs python2Packages.azure-mgmt-policyinsights python2Packages.azure-mgmt-powerbiembedded python2Packages.azure-mgmt-rdbms python2Packages.azure-mgmt-recoveryservices python2Packages.azure-mgmt-recoveryservicesbackup python2Packages.azure-mgmt-redis python2Packages.azure-mgmt-relay python2Packages.azure-mgmt-reservations python2Packages.azure-mgmt-resource python2Packages.azure-mgmt-scheduler python2Packages.azure-mgmt-search python2Packages.azure-mgmt-servicebus python2Packages.azure-mgmt-servicefabric python2Packages.azure-mgmt-signalr python2Packages.azure-mgmt-sql python2Packages.azure-mgmt-storage python2Packages.azure-mgmt-subscription python2Packages.azure-mgmt-trafficmanager python2Packages.azure-mgmt-web python2Packages.azure-servicebus python2Packages.azure-servicefabric python2Packages.azure-servicemanagament-legacy python2Packages.azure-storage-blob python2Packages.uamqp~
> 
> EDIT: Eval still failing, @mwilsoninsight can you PTAL? Thanks!

Yep- had a typo in the metadata that found its way into a few other files due to copy/pasta. It just passed all the checks, so I think it should be all set :crossed_fingers:

nixos-discourse commented:
This pull request has been mentioned on **Nix community**. There might be relevant details there:

https://discourse.nixos.org/t/nixops-travis-ci-test-failure-on-azure-private-ip-pr/2441/12

kalbasit commented:
@GrahamcOfBorg build python2Packages.azure-applicationinsights python2Packages.azure-batch python2Packages.azure-common python2Packages.azure-cosmosdb-nspkg python2Packages.azure-cosmosdb-table python2Packages.azure-datalake-store python2Packages.azure-eventgrid python2Packages.azure-graphrbac python2Packages.azure-keyvault python2Packages.azure-loganalytics python2Packages.azure-mgmt-advisor python2Packages.azure-mgmt-applicationinsights python2Packages.azure-mgmt-authorization python2Packages.azure-mgmt-batch python2Packages.azure-mgmt-batchai python2Packages.azure-mgmt-billing python2Packages.azure-mgmt-cdn python2Packages.azure-mgmt-cognitiveservices python2Packages.azure-mgmt-commerce python2Packages.azure-mgmt-compute python2Packages.azure-mgmt-consumption python2Packages.azure-mgmt-containerinstance python2Packages.azure-mgmt-containerservice python2Packages.azure-mgmt-cosmosdb python2Packages.azure-mgmt-datafactory python2Packages.azure-mgmt-datalake-analytics python2Packages.azure-mgmt-datalake-nspkg python2Packages.azure-mgmt-datalake-store python2Packages.azure-mgmt-datamigration python2Packages.azure-mgmt-devspaces python2Packages.azure-mgmt-devtestlabs python2Packages.azure-mgmt-dns python2Packages.azure-mgmt-eventgrid python2Packages.azure-mgmt-eventhub python2Packages.azure-mgmt-hanaonazure python2Packages.azure-mgmt-iotcentral python2Packages.azure-mgmt-iothub python2Packages.azure-mgmt-iothubprovisioningservices python2Packages.azure-mgmt-keyvault python2Packages.azure-mgmt-loganalytics python2Packages.azure-mgmt-logic python2Packages.azure-mgmt-machinelearningcompute python2Packages.azure-mgmt-managementgroups python2Packages.azure-mgmt-managementpartner python2Packages.azure-mgmt-maps python2Packages.azure-mgmt-marketplaceordering python2Packages.azure-mgmt-media python2Packages.azure-mgmt-monitor python2Packages.azure-mgmt-msi python2Packages.azure-mgmt-network python2Packages.azure-mgmt-notificationhubs python2Packages.azure-mgmt-policyinsights python2Packages.azure-mgmt-powerbiembedded python2Packages.azure-mgmt-rdbms python2Packages.azure-mgmt-recoveryservices python2Packages.azure-mgmt-recoveryservicesbackup python2Packages.azure-mgmt-redis python2Packages.azure-mgmt-relay python2Packages.azure-mgmt-reservations python2Packages.azure-mgmt-resource python2Packages.azure-mgmt-scheduler python2Packages.azure-mgmt-search python2Packages.azure-mgmt-servicebus python2Packages.azure-mgmt-servicefabric python2Packages.azure-mgmt-signalr python2Packages.azure-mgmt-sql python2Packages.azure-mgmt-storage python2Packages.azure-mgmt-subscription python2Packages.azure-mgmt-trafficmanager python2Packages.azure-mgmt-web python2Packages.azure-servicebus python2Packages.azure-servicefabric python2Packages.azure-servicemanagament-legacy python2Packages.azure-storage-blob python2Packages.uamqp

mwilsoninsight commented:
Blerg- I see a failure with python2. I've only built with 3.7 so far, so tomorrow I'll start building for 3.6, 3.5, and 2.7 and sort out any failures :salute:

kalbasit commented:
@mwilsoninsight thank you for working on this! Please ping me when it's ready for another review.

mwilsoninsight commented:
> @mwilsoninsight thank you for working on this! Please ping me when it's ready for another review.

@kalbasit with today's work, it builds successfully against python2.7.

python3.6 threw a deprecation warning while building pylint =/

If context helps, I use a build.nix file with each version of python made using `withPackages (ps: [<each of these packages>])` for testing builds/installs

3.5 threw up with the error that it didn't support contextvars-2.4 (which I don't think any of these modules reference...)

FRidh commented:
Thanks for the PR. Please have a look also at https://github.com/NixOS/nixpkgs/pull/49657#issuecomment-435567034. How are we going to maintain this large set of packages?

FRidh commented:
What's up with removing bytecode? That should not be necessary as we create it during build time. Or was this bytecode part of the sdist?

mwilsoninsight commented:
@FRidh the `rm` statements were necessary to get the modules to build without complaining about colliding files.

These colliding files were of the same path but respective to azure-common and <colliding package>. The files in question were .py and .pyc files that existed in both locations.

The error I encountered primarily regarded namespacing concerns, so I figured that because `azure-common` was a propogatedBuildInput for many packages, that it may be creating the same file in those various locations throughout the nix store, but that the existence of the same files in those libs causes python confusion during namespace declaration.

To be totally honest, I'm not a Pythonista by trade, so I'm doing ~some~ major guess-work here.

FRidh commented:
When did you encounter the collisions, when you used `python.withPackages`?

mwilsoninsight commented:
I encountered it when I used `nix-env -if build.nix`

build.nix (swapping out XY for whichever python version I want to test):
```
with import /home/mwilson/nixpkgs {};

let
  azpkgs =
    ps: with ps;
    [ <each>
      <of>
      <these>
      <packages>
      ...
    ];

  withAzurePackages =
    pythonVersion:
    pythonVersion.withPackages azpkgs;

  self = {
    py37 = withAzurePackages python37;
    py36 = withAzurePackages python36;
    py35 = withAzurePackages python35;
    py27 = withAzurePackages python27;
  };
in
  self.pyXY
```

mwilsoninsight commented:
With certain packages, they would not build because a file in it's lib path collided with the same file in azure-common's lib path (or similar, but azure-common was by _far_ the most frequent to collide with)

I assume I could have just written the same lines in azure-common, but I thought that may not work due to the whole propagatedBuildInput

mwilsoninsight commented:
@FRidh as for:
> Please have a look also at #49657 (comment). How are we going to maintain this large set of packages?

I hadn't given it thought previously- at least no more than, "I should really update these to all use pip once Microsoft fixes the pip release to include code changes from at least ~Feb. 2018". Beyond that, I was planning on revisiting these and updating them as new releases are made to pip for the official repository.

If I'm to maintain these going forward, I'll likely set up a small home-brewed tool for this specific set of packages that will periodically (or more likely, ad-hoc) scrape the pip metadata for changes (values from `curl https://pypi.org/pypi/<package-name>/json | jq` compared to values in `nixpkgs/pkgs/development/python-modules/<package-name>/default.nix`) and then handle them individually as they crop up.

This only works for packages in pip that contain current-code tho- there are still a few pulled from GitHub to get past the Feb 2018 wheel compatibility update.

The tool mentioned in the comment you linked to looked promising, but the links on that page lead to 404s at the moment, so I'm not sure what this `generated.nix` would be. There's still a lot I don't know about nix[pkgs] (and python), but I'm eager to maintain Azure support in NixOps, so I'm open to pretty much anything =]

FRidh commented:
That utility has moved to https://github.com/nix-community/nixpkgs-pytools/.

mwilsoninsight commented:
I just tried the following:
- cloned [the tool](https://github.com/nix-community/nixpkgs-pytools/)
- `cd nixpkgs-pytools`
- `nix-shell`
- `cd ~/nixpkgs/pkgs/development/python-modules/azure-applicationinsights`
- `mv default.nix temp`
- `python-package-init --version 0.1.0 azure-applicationinsights`
- edited the resultant `default.nix` to include `extension="zip";`, `doCheck = false;` (I think I'm missing some dependencies for the tests with applicationinsights- either way, they're failing at the moment, so I'm just turning them off to see if it'll even build), and `maintainers`
- edited `~/build.nix` to only include azure-applicationinsights in the list of packages
- `nix-env -if ~/build.nix`

Result:
`collision between '/nix/store/<sha>-python3.7-azure-common-1.1.19/lib/python3.7/site-packages/azure/__pycache__/__init__.cpython-37.pyc' and '/nix/store/<sha>-python3.7-azure-applicationinsights-0.1.0/lib/python3.7/site-packages/azure/__pycache__/__init__.cpython-37.pyc'`

mwilsoninsight commented:
I've been wondering if it has anything to do with this excerpt from the `default.nix` for azure-common, but I don't know enough about python namespacing to tell for sure:
```
postInstall = ''
    echo "__import__('pkg_resources').declare_namespace(__name__)" >> "$out/lib/${python.libPrefix}"/site-packages/azure/__init__.py
  '';
```

FRidh commented:
Haven't tried yet, but pretty sure this only occurs when creating the `python.withPackages` env, and that is documented behaviour; use `python.buildEnv` instead.

mwilsoninsight commented:
> Haven't tried yet, but pretty sure this only occurs when creating the `python.withPackages` env, and that is documented behaviour; use `python.buildEnv` instead.

Can do. Not too excited to redo all of this work, but it'll be worth it in the long run.

Many thanks for pointing me to nixpkgs-pytools as well- should make the work go a bit smoother =]

FRidh commented:
You don't have to redo anything. It's just that these `rm` commands likely need to be removed.

mwilsoninsight commented:
> You don't have to redo anything. It's just that these `rm` commands likely need to be removed.

Fair enough- taking a look now 👍

mwilsoninsight commented:
@FRidh there's a strange inconsistency with the package set currently available from pip that is labeled as azure-4.0.0 and the src code available from GitHub [tagged as azure_4.0.0](https://github.com/Azure/azure-sdk-for-python/tree/azure_4.0.0)- not only do some packages' versions differ between the two (servicemanagement-legacy is 0.20.6 via pip, but .7 via the tagged src code...), but they also strip the tests from the src.

This means both that testing is disabled and that, while most packages are being fetched from pip, some are being fetched from GitHub- introducing an inconsistency in module definition.

If this is considered a problem, a solution could be changing each package to be fetched from GitHub rather than pip. @FRidh, do you believe this is worth pursuing?

jonringer commented:
vimplugins does a similar thing to update packages https://github.com/NixOS/nixpkgs/blob/master/pkgs/misc/vim-plugins/update.py 

I would love to have these changes rolled in, is there anything i could to do to help?

I've confirmed that python37 and python27 are able to build using the the script below

python36 is broken due to a pylint issue but #61648 should fix that once it merges.

python35 is broken due to:
```bash
$ nix-build testall.nix -A py35
error: contextvars-2.4 not supported for interpreter python3.5
(use '--show-trace' to show detailed location information)
```

testall.nix
```nix
with import ./. {};

let
  azpkgs =
    ps: with ps;
    [
      azure-applicationinsights
      azure-batch
      azure-common
      azure-cosmosdb-nspkg
      azure-cosmosdb-table
      azure-datalake-store
      azure-eventgrid
      azure-graphrbac
      azure-keyvault
      azure-loganalytics
      azure-mgmt-advisor
      azure-mgmt-applicationinsights
      azure-mgmt-authorization
      azure-mgmt-batch
      azure-mgmt-batchai
      azure-mgmt-billing
      azure-mgmt-cdn
      azure-mgmt-cognitiveservices
      azure-mgmt-commerce
      azure-mgmt-compute
      azure-mgmt-consumption
      azure-mgmt-containerinstance
      azure-mgmt-containerservice
      azure-mgmt-cosmosdb
      azure-mgmt-datafactory
      azure-mgmt-datalake-analytics
      azure-mgmt-datalake-nspkg
      azure-mgmt-datalake-store
      azure-mgmt-datamigration
      azure-mgmt-devspaces
      azure-mgmt-devtestlabs
      azure-mgmt-dns
      azure-mgmt-eventgrid
      azure-mgmt-eventhub
      azure-mgmt-hanaonazure
      azure-mgmt-iotcentral
      azure-mgmt-iothub
      azure-mgmt-iothubprovisioningservices
      azure-mgmt-keyvault
      azure-mgmt-loganalytics
      azure-mgmt-logic
      azure-mgmt-machinelearningcompute
      azure-mgmt-managementgroups
      azure-mgmt-managementpartner
      azure-mgmt-maps
      azure-mgmt-marketplaceordering
      azure-mgmt-media
      azure-mgmt-monitor
      azure-mgmt-msi
      azure-mgmt-network
      azure-mgmt-notificationhubs
      azure-mgmt-policyinsights
      azure-mgmt-powerbiembedded
      azure-mgmt-rdbms
      azure-mgmt-recoveryservices
      azure-mgmt-recoveryservicesbackup
      azure-mgmt-redis
      azure-mgmt-relay
      azure-mgmt-reservations
      azure-mgmt-resource
      azure-mgmt-scheduler
      azure-mgmt-search
      azure-mgmt-servicebus
      azure-mgmt-servicefabric
      azure-mgmt-signalr
      azure-mgmt-sql
      azure-mgmt-storage
      azure-mgmt-subscription
      azure-mgmt-trafficmanager
      azure-mgmt-web
      azure-servicebus
      azure-servicefabric
      azure-storage-blob
    ];

  withAzurePackages =
    pythonVersion:
    pythonVersion.withPackages azpkgs;

in
  {
    py37 = withAzurePackages python37;
    py36 = withAzurePackages python36;
    py35 = withAzurePackages python35;
    py27 = withAzurePackages python27;
  }
```
bump @kalbasit  on your changes
bump @FRidh  on reponse to mwilson's question

jonringer commented:
I did some more digging, looks like most of the packages have a reliance on contextvars, which is incompatible with python3.5. Not sure if it's worth the effort to find the most recent packages that don't require contextvars and freeze them for python35.... seeing as this package set has been neglected for a while now.

mwilsoninsight commented:
@jonringer thanks for the link to vim's solution to this issue! I'll look into getting something similar set up down the line (possibly when `azure-5.0.0` comes along)

I also got the same results with building against various python versions: 2.7 and 3.7 being the viable interpreters. I think both the py35 and py36 issues are acceptable due to them being errors in different packages, but that's just my opinion. =]

I spoke with @lmazuel about the packaging of `azure-4.0.0` this past week- here's what I understand:
- the release of `wheel-0.31.0` broke the build process for _all_ packages under `azure` (the .whl still works tho, so the affected user base was minimal)
- `azure-4.0.0` supposedly includes all code that fixes this issue as part of its release, but `azure-servicemanagement-legacy-0.27.0` was missed
- this means that even if GitHub src were used, for a subset of packages, a different commit would be necessary
- given the fix is to alter the `azure_4.0.0` tag to include code that would compile against `wheel-0.31.0`, @lmazuel's suggestion was instead to downgrade the `wheel` version during the build process

Given this information, I don't think I'm going to pursue the tests available from GitHub src until `azure-5.0.0` is released (and _hopefully_ doesn't have any missed releases that fail to compile from src).

Instead, I'll be working this week to remove the postInstall steps from my commits and just take the pypi src where possible (and make sure I have accounted for all of the dependencies in this pkg set).

lmazuel commented:
@mwilsoninsight I just checked the `azure_4.0.0` tag, and I can release the 0.20.7 as it was in that tag if it helps. There is no difference with 0.20.6, the testsuite was not compatible with Python 3.7, and when I updated it and I updated the version number by habit. That should allow you to package `azure_4.0.0` "as-is".
For wheel, there is nothing I can really do :(. 
Finally solution indeed would be to do azure 5.0.0, but timeline is incertain right now for this.

mwilsoninsight commented:
> @mwilsoninsight I just checked the `azure_4.0.0` tag, and I can release the 0.20.7 as it was in that tag if it helps. There is no difference with 0.20.6, the testsuite was not compatible with Python 3.7, and when I updated it and I updated the version number by habit. That should allow you to package `azure_4.0.0` "as-is".
> For wheel, there is nothing I can really do :(.
> Finally solution indeed would be to do azure 5.0.0, but timeline is incertain right now for this.

No worries!

Unfortunately this creates a damned-if-you-do-damned-if-you-don't situation for including these modules in nixpkgs (nix peeps, feel free to correct me on this).
As far as I see it our options currently are:
- we keep fetching _some_ src from GitHub and _some_ src from pip
  - this now means we're treating packages uniquely- not a _huge_ concern, but consistency allows for smoother automation (and, arguably, conceptual understanding)
  - we now have to be sure the GitHub src matches a pip release or, more likely, that it at least builds
  - on the upside, this is largely already done, so the updates/tweaks are arguably minor
- ~we fetch everything from pip~
  - the src doesn't build _as distributed_ (and yes, we could downgrade the version of `wheel`, but then we're right back to the consistency concern)
  - It would also require somehow using [./pkgs/top-level/python-packages.nix](https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/python-packages.nix) or [./pkgs/development/python-modules/wheel/default.nix](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/wheel/default.nix) to override the version of `wheel` available to the build environment
    - I need some advice on how to do this if that's the route we'd like to go down
- we fetch everything from GitHub
  - this is the most attractive option IMHO:
    - we can preserve the tests
    - given a single tag, in which all modules build with `wheel-latest`, we can build all modules with minimal disk space usage
    - automation is consistent for all packages (for dependencies: providing GitHub src is tagged in a straight-forward manner and one of those tags matches a current pip release (I believe that was the case for most/all of the ones I came across))
  - however, 4.0.0 has a very small, yet very frustrating issue in which one or so packages don't build with `wheel-latest`, so _at the moment_, we'd be right back to the same consistency concern we have with fetching from pip

IMO, it all ends up being a wash in the current state of things, so I feel it's best to make the smallest amount of changes for the time being and patiently await the next `azure-x.y.z` tag to make the switch to fetchFromGitHub. =]

mwilsoninsight commented:
@kalbasit I think this is ready to merge (providing the extant 3.5 (pylint) and 3.6 (contextvars) errors are acceptable) =]

@FRidh, I think @jonringer's suggestion of deriving something from vim's python update script is a reasonable solution.

Originally I was planning on using these updates to revamp the Azure backend to NixOps, but coming up with the update script first may be more prudent.
Any Python folks willing to contribute in that regard would be _extremely_ welcome- it'll be on my back-burner otherwise.

jonringer commented:
@mwilsoninsight I'm not a pythonista, but my job is azure related (work on AzureML), so I would definitely like to have something like this as it's going to be my future for a while.

mwilsoninsight commented:
@lmazuel there have been a few releases to sub-modules since I originally opened this issue- as such, the most recent round of force-pushed changes includes most of them.

This ended up resolving the failed build against `wheel-0.31.0` for everything except the following (current pip release still won't build out-of-the-box):
`azure-servicemanagement-legacy`
`azure-mgmt-billing`
`azure-mgmt-commerce`

Those are instead being fetched from GitHub.

Also, `azure-cosmosdb-table-1.0.5` is the latest of its versions, but it requires `azure-storage-common~=1.1`- this means I left off the v2.0.1 upgrades of azure-storage-{common,file,queue,blob} at the moment so that the entire feature set of `azure-4.0.0` could be included without breaking builds.

jonringer commented on pkgs/development/python-modules/azure-servicemanagement-legacy/default.nix:
```diff
@@ -1,31 +1,39 @@
 , azure-common
 , requests
 , python
+, isPy3k
```
This doesn't get used in the expression :)

mwilsoninsight replied:
XD
aaaaaand that's _exactly_ why I need to automate this biz...

`python` doesn't get used either, now that the `rm` statements are gone.

Just pushed a rebase to fix it. Many thanks! =]

jonringer commented:
@mwilsoninsight we may want to remove lines like this for python3:

azure-mgmt-storage
```
  postInstall = ''
    echo "__import__('pkg_resources').declare_namespace(__name__)" >> "$out/lib/${python.libPrefix}"/site-packages/azure/__init__.py
    echo "__import__('pkg_resources').declare_namespace(__name__)" >> "$out/lib/${python.libPrefix}"/site-packages/azure/mgmt/__init__.py
  '';
```

Due to how https://www.python.org/dev/peps/pep-0420/ works, if we have an __init__.py at that directory path then the interpreter will load it as a regular module, this is a problem because all the packages are supposed to be loaded under a namespace (think directory). So, it's not an issue when you load and use one package, but if you have many (and some packages of dependencies on sibling packages) it's going to load only the first in the python path that matches, and the others get "shadowed". 

Removing the __init__.py's has the intended effect, in which python3 will attempt to find the package by traversing all the paths in PYTHONPATH. Since it won't find a "module"(__init.py) at something like "azure" it will continue traversing.

mwilsoninsight commented:
> @mwilsoninsight we may want to remove lines like this for python3:

Cool beans! I was wondering if that was a possibility for some of these, but I didn't quite know how class-loading would work in the end, so I left those as they were added by @olcai.

So something like the following for all of the packages with a `postInstall` that writes a (or a few) `__init__.py` files?:
```
postInstall = if isPy3k then "" else ''
  echo $import_declare_namespace_statement >> $desired_directory/__init__.py
'';
```

lmazuel commented:
Hey @mwilsoninsight , @jonringer is right, if you sure you use Python 3, kill these files :). I actually have a [conditional dependency](https://github.com/Azure/azure-sdk-for-python/blob/4d9560cfd519ee60667f3cc2f5295a58c18625db/azure-mgmt-compute/setup.py#L86) to do NOT install them on Python 3.

If you really have to, for some reasons, don't use this prehistoric syntax but this:
```python
__path__ = __import__('pkgutil').extend_path(__path__, __name__)
```
https://github.com/pypa/sample-namespace-packages

mwilsoninsight commented:
> Hey @mwilsoninsight , @jonringer is right, if you sure you use Python 3, kill these files :). I actually have a [conditional dependency](https://github.com/Azure/azure-sdk-for-python/blob/4d9560cfd519ee60667f3cc2f5295a58c18625db/azure-mgmt-compute/setup.py#L86) to do NOT install them on Python 3.

Sweet!

With this confirmation, I've included the `if isPy3k then "" else ''echo $the_thing >> $the_place/__init__.py'';` fix in the latest rebase- meaning any builds against Python 3 will not include the extraneous `__init__.py` files. =]

jonringer commented:
hmm, how did this get labeled as erlang and nixos?

mwilsoninsight commented:
> hmm, how did this get labeled as erlang and nixos

ugh.. bad git history... I accidentally pulled in some commits from other projects by trying to update the future branch- should be fixed now, but ofborg tends to lag behind by a few minutes

jonringer commented:
@mwilsoninsight looks like your update script may already exist https://github.com/NixOS/nixpkgs/pull/50983

mwilsoninsight commented:
@jonringer that looks like it may do the trick!

I'm taking a look into it :+1:

jonringer commented:
well, even if some packages get "left behind", i think infrequently updating manually shouldn't be too bad. I think a lot of pain with this PR is that there's only been like... 5ish of these ever, out of the 100+ in the pypi repo.

jonringer commented:
@FRidh @mwilsoninsight @kalbasit I think this is ready to merge.

Can't wait to drop my huge overlay for these changes. lol

FRidh commented:
The referenced update script only updates the version and hash, but not any of the dependencies.

I'm a bit skeptical to including this in Nixpkgs without a close-to-automated way of maintaining it. How often does upstream update these packages, and change dependencies? The good thing is that it doesn't depend much on other, non-azure, packages so it's less likely to break in that regard.

As users, do you "expect" a working version on master, or only on stable?

jonringer commented:
@FRidh 
> How often does upstream update these packages

Per package, not too often, but they seem to have independent release cycles per package. Some are more "stable" in feature scope and others are "very active"

> ..., and change dependencies? 

Rarely ever.

Most of the packages are a vertical slice of the entire Azure platform, the majority of them will rely only on [ msrest msrestazure azure-common~=1.1 ] but no "web" of sibling dependencies. Azure-common is the only real concerning dependency, but it hasn't had a minor bump for [3 years now](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/core/azure-common/HISTORY.rst#user-content-110-2016-03-04).

@lmazuel would be better equipped to answer how likely there will be issues with changing dependencies in the future.

nixos-discourse commented:
This pull request has been mentioned on **Nix community**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review-may-2019/3032/10

vanschelven commented:
I wonder how this PR relates to the file `pkgs/development/python-modules/azure/default.nix`. I recently tried to install that package, but it failed. I believe (but did not thoroughly research) that it is a package that has basically been broken since its initial creation. In any case, the goal of that package is:

> This package does not contain any code in itself. It installs a set of packages that provide Microsoft Azure functionality.

But this is not reflected at all in its nix expression. I would think such an expression would contain many `propagatedBuildInputs`.

Back to the present PR: since this PR is an "Azure SDK update", perhaps we can use it as an opportunity to fix the mentioned umbrella-package (include all the referenced packages). Alternatively, we might drop the umbrella package altogether.

jonringer commented:
@vanschelven correct, the pythonPackages.azure is currently broken. The setup.py for it can be found https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/core/azure/setup.py and you can see that it just exports a common selection of the azure namespace packages, which aren't expressed in `pkgs/development/python-modules/azure/default.nix`

I think the umbrella package is fine, and it aligns with what you would expect with what you would expect to be there on pypi, it just doesn't work while the dependencies don't exist. Which this PR should remedy.

jonringer commented:
bump @kalbasit  @FRidh 

The core parts of these namespace packages have been fairly stable for the past few years. And seeing as this will be a part of my work life for the foreseeable future, I'll be sure to keep it up to date. 

@mwilsoninsight if you want, you can add me as a maintainer, or I'll just add myself if there are updates to certain packages. I'm sure iterating over 80 commits isn't the most fun.

FRidh commented:
All the `lib.optional` need to be replaced by `lib.optionals` because they're lists that get appended. What happens now is that a list gets inserted into a list and appended to a list resulting in nested lists.

At this point, don't bother anymore with the commit per package rule, I don't think it's worth it, especially when adding this many closely-related packages.

Python 2.7 and 3.7 is what we "support". 3.5 and 3.6, while installable, we don't build on Hydra.

After these changes and a rebase it can be merged.

jonringer commented on pkgs/development/python-modules/azure-mgmt-batch/default.nix:
```diff
@@ -0,0 +1,38 @@
+    msrest
+    msrestazure
+    azure-common
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented on pkgs/development/python-modules/azure-mgmt-cdn/default.nix:
```diff
@@ -0,0 +1,38 @@
+    msrest
+    msrestazure
+    azure-common
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented on pkgs/development/python-modules/azure-mgmt-containerinstance/default.nix:
```diff
@@ -0,0 +1,38 @@
+    msrest
+    msrestazure
+    azure-common
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented on pkgs/development/python-modules/azure-mgmt-containerservice/default.nix:
```diff
@@ -0,0 +1,38 @@
+    msrest
+    msrestazure
+    azure-common
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented on pkgs/development/python-modules/azure-mgmt-cosmosdb/default.nix:
```diff
@@ -0,0 +1,38 @@
+    msrest
+    msrestazure
+    azure-common
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented on pkgs/development/python-modules/azure-mgmt-datafactory/default.nix:
```diff
@@ -0,0 +1,38 @@
+    msrest
+    msrestazure
+    azure-common
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented on pkgs/development/python-modules/azure-mgmt-datamigration/default.nix:
```diff
@@ -0,0 +1,38 @@
+    msrest
+    msrestazure
+    azure-common
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented on pkgs/development/python-modules/azure-mgmt-devtestlabs/default.nix:
```diff
@@ -0,0 +1,38 @@
+    msrest
+    msrestazure
+    azure-common
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented on pkgs/development/python-modules/azure-mgmt-eventgrid/default.nix:
```diff
@@ -0,0 +1,38 @@
+    msrest
+    msrestazure
+    azure-common
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented on pkgs/development/python-modules/azure-cosmosdb-table/default.nix:
```diff
@@ -0,0 +1,39 @@
+    azure-common
+    azure-storage-common
+    azure-cosmosdb-nspkg
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented on pkgs/development/python-modules/azure-mgmt-advisor/default.nix:
```diff
@@ -0,0 +1,38 @@
+    msrest
+    msrestazure
+    azure-common
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented on pkgs/development/python-modules/azure-mgmt-applicationinsights/default.nix:
```diff
@@ -0,0 +1,38 @@
+    msrest
+    msrestazure
+    azure-common
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented on pkgs/development/python-modules/azure-mgmt-authorization/default.nix:
```diff
@@ -0,0 +1,38 @@
+    msrest
+    msrestazure
+    azure-common
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented on pkgs/development/python-modules/azure-mgmt-eventhub/default.nix:
```diff
@@ -0,0 +1,38 @@
+    msrest
+    msrestazure
+    azure-common
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented on pkgs/development/python-modules/azure-mgmt-hanaonazure/default.nix:
```diff
@@ -0,0 +1,38 @@
+    msrest
+    msrestazure
+    azure-common
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented on pkgs/development/python-modules/azure-mgmt-iotcentral/default.nix:
```diff
@@ -0,0 +1,38 @@
+    msrest
+    msrestazure
+    azure-common
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented on pkgs/development/python-modules/azure-mgmt-iothub/default.nix:
```diff
@@ -0,0 +1,38 @@
+    msrest
+    msrestazure
+    azure-common
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented on pkgs/development/python-modules/azure-mgmt-managementgroups/default.nix:
```diff
@@ -0,0 +1,38 @@
+    msrest
+    msrestazure
+    azure-common
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented on pkgs/development/python-modules/azure-mgmt-marketplaceordering/default.nix:
```diff
@@ -0,0 +1,38 @@
+    msrest
+    msrestazure
+    azure-common
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented on pkgs/development/python-modules/azure-mgmt-media/default.nix:
```diff
@@ -0,0 +1,38 @@
+    msrest
+    msrestazure
+    azure-common
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented on pkgs/development/python-modules/azure-mgmt-monitor/default.nix:
```diff
@@ -0,0 +1,38 @@
+    msrest
+    msrestazure
+    azure-common
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented on pkgs/development/python-modules/azure-mgmt-policyinsights/default.nix:
```diff
@@ -0,0 +1,38 @@
+    msrest
+    msrestazure
+    azure-common
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented on pkgs/development/python-modules/azure-mgmt-rdbms/default.nix:
```diff
@@ -0,0 +1,38 @@
+    msrest
+    msrestazure
+    azure-common
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented on pkgs/development/python-modules/azure-mgmt-recoveryservices/default.nix:
```diff
@@ -0,0 +1,38 @@
+    msrest
+    msrestazure
+    azure-common
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented on pkgs/development/python-modules/azure-mgmt-redis/default.nix:
```diff
@@ -0,0 +1,38 @@
+    msrest
+    msrestazure
+    azure-common
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented on pkgs/development/python-modules/azure-mgmt-reservations/default.nix:
```diff
@@ -0,0 +1,38 @@
+    msrest
+    msrestazure
+    azure-common
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented on pkgs/development/python-modules/azure-mgmt-servicebus/default.nix:
```diff
@@ -0,0 +1,38 @@
+    msrest
+    msrestazure
+    azure-common
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented on pkgs/development/python-modules/azure-mgmt-sql/default.nix:
```diff
@@ -0,0 +1,38 @@
+    msrest
+    msrestazure
+    azure-common
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented on pkgs/development/python-modules/azure-mgmt-subscription/default.nix:
```diff
@@ -0,0 +1,38 @@
+    msrest
+    msrestazure
+    azure-common
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented on pkgs/development/python-modules/azure-mgmt-trafficmanager/default.nix:
```diff
@@ -0,0 +1,38 @@
+    msrest
+    msrestazure
+    azure-common
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented on pkgs/development/python-modules/azure-mgmt-web/default.nix:
```diff
@@ -0,0 +1,38 @@
+    msrest
+    msrestazure
+    azure-common
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented on pkgs/development/python-modules/azure-servicebus/default.nix:
```diff
@@ -0,0 +1,38 @@
+    uamqp
+    azure-common
+    msrestazure
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented on pkgs/development/python-modules/uamqp/default.nix:
```diff
@@ -0,0 +1,42 @@
+    openssl
+    certifi
+    six
+  ] ++ lib.optional (!isPy3k) [
```
```suggestion
  ] ++ lib.optionals (!isPy3k) [
```

jonringer commented:
@FRidh original author hasn't been seen in the past 3-ish weeks, is it alright if i hijack this PR? not sure what the protocol for this would be.

If i do hijack it, i would just do another commit where i would add myself as a maintainer and apply the `optionals` changes

kalbasit commented:
I fixed all the `lib.optional -> lib.optionals` and rebased the PR over master. I tested locally on my NixOS and I'm able to build all packages over on Python2 and Python3. I'm going to give ofborg a try, and merge if it works. @FRidh please let me know if you have any reservations against merging this PR.

@GrahamcOfBorg build python2Packages.azure-applicationinsights python2Packages.azure-batch python2Packages.azure-common python2Packages.azure-cosmosdb-nspkg python2Packages.azure-cosmosdb-table python2Packages.azure-datalake-store python2Packages.azure-eventgrid python2Packages.azure-graphrbac python2Packages.azure-keyvault python2Packages.azure-loganalytics python2Packages.azure-mgmt-advisor python2Packages.azure-mgmt-applicationinsights python2Packages.azure-mgmt-authorization python2Packages.azure-mgmt-batch python2Packages.azure-mgmt-batchai python2Packages.azure-mgmt-billing python2Packages.azure-mgmt-cdn python2Packages.azure-mgmt-cognitiveservices python2Packages.azure-mgmt-commerce python2Packages.azure-mgmt-common python2Packages.azure-mgmt-compute python2Packages.azure-mgmt-consumption python2Packages.azure-mgmt-containerinstance python2Packages.azure-mgmt-containerservice python2Packages.azure-mgmt-cosmosdb python2Packages.azure-mgmt-datafactory python2Packages.azure-mgmt-datalake-analytics python2Packages.azure-mgmt-datalake-nspkg python2Packages.azure-mgmt-datalake-store python2Packages.azure-mgmt-datamigration python2Packages.azure-mgmt-devspaces python2Packages.azure-mgmt-devtestlabs python2Packages.azure-mgmt-dns python2Packages.azure-mgmt-eventgrid python2Packages.azure-mgmt-eventhub python2Packages.azure-mgmt-hanaonazure python2Packages.azure-mgmt-iotcentral python2Packages.azure-mgmt-iothub python2Packages.azure-mgmt-iothubprovisioningservices python2Packages.azure-mgmt-keyvault python2Packages.azure-mgmt-loganalytics python2Packages.azure-mgmt-logic python2Packages.azure-mgmt-machinelearningcompute python2Packages.azure-mgmt-managementgroups python2Packages.azure-mgmt-managementpartner python2Packages.azure-mgmt-maps python2Packages.azure-mgmt-marketplaceordering python2Packages.azure-mgmt-media python2Packages.azure-mgmt-monitor python2Packages.azure-mgmt-msi python2Packages.azure-mgmt-network python2Packages.azure-mgmt-notificationhubs python2Packages.azure-mgmt-nspkg python2Packages.azure-mgmt-policyinsights python2Packages.azure-mgmt-powerbiembedded python2Packages.azure-mgmt-rdbms python2Packages.azure-mgmt-recoveryservices python2Packages.azure-mgmt-recoveryservicesbackup python2Packages.azure-mgmt-redis python2Packages.azure-mgmt-relay python2Packages.azure-mgmt-reservations python2Packages.azure-mgmt-resource python2Packages.azure-mgmt-scheduler python2Packages.azure-mgmt-search python2Packages.azure-mgmt-servicebus python2Packages.azure-mgmt-servicefabric python2Packages.azure-mgmt-signalr python2Packages.azure-mgmt-sql python2Packages.azure-mgmt-storage python2Packages.azure-mgmt-subscription python2Packages.azure-mgmt-trafficmanager python2Packages.azure-mgmt-web python2Packages.azure-nspkg python2Packages.azure-servicebus python2Packages.azure-servicefabric python2Packages.azure-servicemanagement-legacy python2Packages.azure-storage-blob python2Packages.msrest python2Packages.uamqp

@GrahamcOfBorg build python3Packages.azure-applicationinsights python3Packages.azure-batch python3Packages.azure-common python3Packages.azure-cosmosdb-nspkg python3Packages.azure-cosmosdb-table python3Packages.azure-datalake-store python3Packages.azure-eventgrid python3Packages.azure-graphrbac python3Packages.azure-keyvault python3Packages.azure-loganalytics python3Packages.azure-mgmt-advisor python3Packages.azure-mgmt-applicationinsights python3Packages.azure-mgmt-authorization python3Packages.azure-mgmt-batch python3Packages.azure-mgmt-batchai python3Packages.azure-mgmt-billing python3Packages.azure-mgmt-cdn python3Packages.azure-mgmt-cognitiveservices python3Packages.azure-mgmt-commerce python3Packages.azure-mgmt-common python3Packages.azure-mgmt-compute python3Packages.azure-mgmt-consumption python3Packages.azure-mgmt-containerinstance python3Packages.azure-mgmt-containerservice python3Packages.azure-mgmt-cosmosdb python3Packages.azure-mgmt-datafactory python3Packages.azure-mgmt-datalake-analytics python3Packages.azure-mgmt-datalake-nspkg python3Packages.azure-mgmt-datalake-store python3Packages.azure-mgmt-datamigration python3Packages.azure-mgmt-devspaces python3Packages.azure-mgmt-devtestlabs python3Packages.azure-mgmt-dns python3Packages.azure-mgmt-eventgrid python3Packages.azure-mgmt-eventhub python3Packages.azure-mgmt-hanaonazure python3Packages.azure-mgmt-iotcentral python3Packages.azure-mgmt-iothub python3Packages.azure-mgmt-iothubprovisioningservices python3Packages.azure-mgmt-keyvault python3Packages.azure-mgmt-loganalytics python3Packages.azure-mgmt-logic python3Packages.azure-mgmt-machinelearningcompute python3Packages.azure-mgmt-managementgroups python3Packages.azure-mgmt-managementpartner python3Packages.azure-mgmt-maps python3Packages.azure-mgmt-marketplaceordering python3Packages.azure-mgmt-media python3Packages.azure-mgmt-monitor python3Packages.azure-mgmt-msi python3Packages.azure-mgmt-network python3Packages.azure-mgmt-notificationhubs python3Packages.azure-mgmt-nspkg python3Packages.azure-mgmt-policyinsights python3Packages.azure-mgmt-powerbiembedded python3Packages.azure-mgmt-rdbms python3Packages.azure-mgmt-recoveryservices python3Packages.azure-mgmt-recoveryservicesbackup python3Packages.azure-mgmt-redis python3Packages.azure-mgmt-relay python3Packages.azure-mgmt-reservations python3Packages.azure-mgmt-resource python3Packages.azure-mgmt-scheduler python3Packages.azure-mgmt-search python3Packages.azure-mgmt-servicebus python3Packages.azure-mgmt-servicefabric python3Packages.azure-mgmt-signalr python3Packages.azure-mgmt-sql python3Packages.azure-mgmt-storage python3Packages.azure-mgmt-subscription python3Packages.azure-mgmt-trafficmanager python3Packages.azure-mgmt-web python3Packages.azure-nspkg python3Packages.azure-servicebus python3Packages.azure-servicefabric python3Packages.azure-servicemanagement-legacy python3Packages.azure-storage-blob python3Packages.msrest python3Packages.uamqp

kalbasit commented:
@GrahamcOfBorg build python3Packages.uamqp python3Packages.azure-servicebus

kalbasit commented:
Darwin issue fixed, retrying borg.

<details>
<summary>ofborg command</summary>

@GrahamcOfBorg build python2Packages.azure-applicationinsights python2Packages.azure-batch python2Packages.azure-common python2Packages.azure-cosmosdb-nspkg python2Packages.azure-cosmosdb-table python2Packages.azure-datalake-store python2Packages.azure-eventgrid python2Packages.azure-graphrbac python2Packages.azure-keyvault python2Packages.azure-loganalytics python2Packages.azure-mgmt-advisor python2Packages.azure-mgmt-applicationinsights python2Packages.azure-mgmt-authorization python2Packages.azure-mgmt-batch python2Packages.azure-mgmt-batchai python2Packages.azure-mgmt-billing python2Packages.azure-mgmt-cdn python2Packages.azure-mgmt-cognitiveservices python2Packages.azure-mgmt-commerce python2Packages.azure-mgmt-common python2Packages.azure-mgmt-compute python2Packages.azure-mgmt-consumption python2Packages.azure-mgmt-containerinstance python2Packages.azure-mgmt-containerservice python2Packages.azure-mgmt-cosmosdb python2Packages.azure-mgmt-datafactory python2Packages.azure-mgmt-datalake-analytics python2Packages.azure-mgmt-datalake-nspkg python2Packages.azure-mgmt-datalake-store python2Packages.azure-mgmt-datamigration python2Packages.azure-mgmt-devspaces python2Packages.azure-mgmt-devtestlabs python2Packages.azure-mgmt-dns python2Packages.azure-mgmt-eventgrid python2Packages.azure-mgmt-eventhub python2Packages.azure-mgmt-hanaonazure python2Packages.azure-mgmt-iotcentral python2Packages.azure-mgmt-iothub python2Packages.azure-mgmt-iothubprovisioningservices python2Packages.azure-mgmt-keyvault python2Packages.azure-mgmt-loganalytics python2Packages.azure-mgmt-logic python2Packages.azure-mgmt-machinelearningcompute python2Packages.azure-mgmt-managementgroups python2Packages.azure-mgmt-managementpartner python2Packages.azure-mgmt-maps python2Packages.azure-mgmt-marketplaceordering python2Packages.azure-mgmt-media python2Packages.azure-mgmt-monitor python2Packages.azure-mgmt-msi python2Packages.azure-mgmt-network python2Packages.azure-mgmt-notificationhubs python2Packages.azure-mgmt-nspkg python2Packages.azure-mgmt-policyinsights python2Packages.azure-mgmt-powerbiembedded python2Packages.azure-mgmt-rdbms python2Packages.azure-mgmt-recoveryservices python2Packages.azure-mgmt-recoveryservicesbackup python2Packages.azure-mgmt-redis python2Packages.azure-mgmt-relay python2Packages.azure-mgmt-reservations python2Packages.azure-mgmt-resource python2Packages.azure-mgmt-scheduler python2Packages.azure-mgmt-search python2Packages.azure-mgmt-servicebus python2Packages.azure-mgmt-servicefabric python2Packages.azure-mgmt-signalr python2Packages.azure-mgmt-sql python2Packages.azure-mgmt-storage python2Packages.azure-mgmt-subscription python2Packages.azure-mgmt-trafficmanager python2Packages.azure-mgmt-web python2Packages.azure-nspkg python2Packages.azure-servicebus python2Packages.azure-servicefabric python2Packages.azure-servicemanagement-legacy python2Packages.azure-storage-blob python2Packages.msrest python2Packages.uamqp

@GrahamcOfBorg build python3Packages.azure-applicationinsights python3Packages.azure-batch python3Packages.azure-common python3Packages.azure-cosmosdb-nspkg python3Packages.azure-cosmosdb-table python3Packages.azure-datalake-store python3Packages.azure-eventgrid python3Packages.azure-graphrbac python3Packages.azure-keyvault python3Packages.azure-loganalytics python3Packages.azure-mgmt-advisor python3Packages.azure-mgmt-applicationinsights python3Packages.azure-mgmt-authorization python3Packages.azure-mgmt-batch python3Packages.azure-mgmt-batchai python3Packages.azure-mgmt-billing python3Packages.azure-mgmt-cdn python3Packages.azure-mgmt-cognitiveservices python3Packages.azure-mgmt-commerce python3Packages.azure-mgmt-common python3Packages.azure-mgmt-compute python3Packages.azure-mgmt-consumption python3Packages.azure-mgmt-containerinstance python3Packages.azure-mgmt-containerservice python3Packages.azure-mgmt-cosmosdb python3Packages.azure-mgmt-datafactory python3Packages.azure-mgmt-datalake-analytics python3Packages.azure-mgmt-datalake-nspkg python3Packages.azure-mgmt-datalake-store python3Packages.azure-mgmt-datamigration python3Packages.azure-mgmt-devspaces python3Packages.azure-mgmt-devtestlabs python3Packages.azure-mgmt-dns python3Packages.azure-mgmt-eventgrid python3Packages.azure-mgmt-eventhub python3Packages.azure-mgmt-hanaonazure python3Packages.azure-mgmt-iotcentral python3Packages.azure-mgmt-iothub python3Packages.azure-mgmt-iothubprovisioningservices python3Packages.azure-mgmt-keyvault python3Packages.azure-mgmt-loganalytics python3Packages.azure-mgmt-logic python3Packages.azure-mgmt-machinelearningcompute python3Packages.azure-mgmt-managementgroups python3Packages.azure-mgmt-managementpartner python3Packages.azure-mgmt-maps python3Packages.azure-mgmt-marketplaceordering python3Packages.azure-mgmt-media python3Packages.azure-mgmt-monitor python3Packages.azure-mgmt-msi python3Packages.azure-mgmt-network python3Packages.azure-mgmt-notificationhubs python3Packages.azure-mgmt-nspkg python3Packages.azure-mgmt-policyinsights python3Packages.azure-mgmt-powerbiembedded python3Packages.azure-mgmt-rdbms python3Packages.azure-mgmt-recoveryservices python3Packages.azure-mgmt-recoveryservicesbackup python3Packages.azure-mgmt-redis python3Packages.azure-mgmt-relay python3Packages.azure-mgmt-reservations python3Packages.azure-mgmt-resource python3Packages.azure-mgmt-scheduler python3Packages.azure-mgmt-search python3Packages.azure-mgmt-servicebus python3Packages.azure-mgmt-servicefabric python3Packages.azure-mgmt-signalr python3Packages.azure-mgmt-sql python3Packages.azure-mgmt-storage python3Packages.azure-mgmt-subscription python3Packages.azure-mgmt-trafficmanager python3Packages.azure-mgmt-web python3Packages.azure-nspkg python3Packages.azure-servicebus python3Packages.azure-servicefabric python3Packages.azure-servicemanagement-legacy python3Packages.azure-storage-blob python3Packages.msrest python3Packages.uamqp

</details>

kalbasit commented:
All my testing are good, merging this PR.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/nixops-travis-ci-test-failure-on-azure-private-ip-pr/2441/13
